src/service/folder_service.rs

```diff
@@ -753,3 +753,106 @@ mod download_folder_tests {
         cleanup();
     }
 }
+
+#[cfg(test)]
+mod tag_inheritance_tests {
+    use crate::{
+        test::{cleanup, create_file_db_entry, create_folder_db_entry, create_tag_folder, init_db_folder},
+        tags::service as tag_service,
+        repository::open_connection,
+        tags::repository as tag_repository,
+    };
+
+    #[test]
+    fn pass_tags_to_children_adds_tags_to_descendant_folders() {
+        init_db_folder();
+        // Create folder hierarchy: top -> middle -> bottom
+        create_folder_db_entry("top", None); // 1
+        create_folder_db_entry("middle", Some(1)); // 2
+        create_folder_db_entry("bottom", Some(2)); // 3
+        
+        // Add tag to top folder
+        create_tag_folder("tag1", 1);
+        
+        // Pass tags to children
+        tag_service::pass_tags_to_children(1).unwrap();
```

Comment:
These tests are misguided. They should test that update_folder in folder service ensures that the tags are inherited. Manually calling tag service and checking the repository doesn't test anything.

Make sure tests are also added for the new pass tags to children function in the tags module